.github/workflows/clang-tidy.yml

```
@@ -48,10 +51,28 @@ jobs:
         env:
           ONEDNN_ACTION: configure
 
-      - name: Build oneDNN
+      - name: Check source files
         run: |
-          .github/automation/x64/build_linters.sh 2>&1 | tee build.log
-          grep -i "warning:" build.log | sort -u
-          grep -q -i "warning:" build.log && exit 1 || true
-        env:
-          ONEDNN_ACTION: build
+          touch source-check.log
+          for file in $(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.(c|cpp)$'); 
+          do
+            if grep -q "$file" "build/compile_commands.json"; then
+              clang-tidy -p build $file 2>&1 | tee -a source-check.log
+            fi
```

A little diagnostic printing may be helpful to understand what's going on.

```suggestion
            if grep -q "$file" "build/compile_commands.json"; then
              echo Analyzing $file
              clang-tidy -p build $file 2>&1 | tee -a source-check.log
            else
              echo Skipped $file as it's not used in x64 OpenMP/OpenCL configuration.
            fi
```